### DISCUSSION TITLE
Criteria for selecting batch samples

### DISCUSSION THREAD

**Discussion by Paeyus** [meta-pytorch/botorch#2352] [General]
I would like to understand the criteria used by the acquisition function (e.g., acquisition score, uncertainty) for selecting sample points in batch optimization. Additionally, is it possible for a user to print out the acquisition score while coding? If there are any published articles on this topic, please recommend them to me. Thank you.

**Comment by saitcakmak** [comment]
Hi @Paeyus. There are two ways batch acquisition optimization is supported in BoTorch. For a typical use case, this is controlled using `sequential` argument to `optimize_acqf`. How this affects the computation of the acquisition value will depend on the particular acquisition function, but generally speaking:
- joint optimization will jointly sample from the posterior on all candidate points & compute the best point (for single objective or the joint hypervolume for multi-objective) over all samples (the q-batch in BoTorch terminology). During optimization, the gradients are propagated to all candidates and they all get optimized jointly.
- sequential optimization will optimize the first point as a regular single candidate. For the subsequent candidates, it will mimic fantasizing from the previously generated candidates. In practice, this is simply achieved by jointly sampling from previously generated & the current candidates, similar to what is done for the joint optimization setting (the acquisition value may be augmented to avoid double counting). The optimizer will only optimize the current candidate and the gradients will not be computed for any previously generated points. 

These two papers have some comparison of the two approaches: 
- https://proceedings.neurips.cc/paper_files/paper/2018/file/498f2c21688f6451d9f5fd09d53edda7-Paper.pdf
- https://arxiv.org/abs/2310.20708
